_posts/2021-07-05-newsletter.md

```diff
@@ -0,0 +1,199 @@
+
+## Tree-sitter
+
+Neovim 0.5 adds **experimental** support for tree-sitter, a library that parses a piece of code into a syntax tree in an incremental and error-resilient way; this means that reparsing that code after an edit is very fast, and parsing error due to, e.g., typos remain localized and do not break parsing further down. This tree can then be efficiently queried to obtain syntax information about the code. This allows for improved and/or faster
+
+- syntax highlighting,
+- code navigation,
+- refactoring,
+- text objects and motions,
+- searching and replacing,
+
+and more.
+
+In addition, tree-sitter allows "injecting" trees for other languages, which makes it easy to support files that contain code in multiple languages. To learn more about tree-sitter, watch [Tree-sitter - A new parsing system for programming tools - Max Brunsfield](https://www.youtube.com/watch?v=Jes3bD6P0To&feature=youtu.be&t=232).
+
+The goal is to replace the current vim regular expression-based syntax with tree-sitter, not only for better and faster syntax highlighting but also for new and improved ways of structured text editing. Currently, Neovim embeds or links to the tree-sitter library and provides a (Lua) API for generating and querying the syntax tree. User-facing features like the ones mentioned above are implemented in plugins like
+
+- [Nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) -- Highlighting, folds, incremental selection.
+- [Playground](https://github.com/nvim-treesitter/playground) -- Utility functions to easily show the parsed tree and run queries against it.
+- [Nvim-treesitter-textobjects](https://github.com/nvim-treesitter/nvim-treesitter-textobjects) -- Better text objects for vim-style motions and operators.
+- [Nvim-refactor](https://github.com/nvim-treesitter/nvim-treesitter-refactor) -- Highlight definitions,  navigation, smart rename.
+- [Architext](https://github.com/vigoux/architext.nvim) -- structural text editing (i.e., syntax-aware search and replace).
+
+More information on Neovim's tree-sitter features can be found in [Nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) and read [`:h treesitter`](https://neovim.io/doc/user/treesitter.html); you should also check out Thomas Vigouroux's [Vimconf.live presentation](https://youtu.be/EReNOGuMBmo?t=515). 
+
+Although these plugins already work well enough in some situations (file types and/or features), there are still a number of bugs and performance regressions that may occur regularly and that need to be addressed before tree-sitter support in Neovim can be declared stable. Note also that enabling tree-sitter based highlighting currently disables the internal regex-based syntax engine _completely_ (which may break other features that rely on it). Fixing these issues and improving the API (which is not stable yet) will be a major focus of the development cycle leading up to the 0.6 release.
```

Comment:
>  Note also that enabling tree-sitter based highlighting currently disables the internal regex-based syntax engine _completely_

with the strong emphasis of "completely" it should be mentioned that is "completely" is still only per filetype.